src/find-options/FindOptionsUtils.ts

```diff
@@ -94,6 +96,11 @@ export class FindOptionsUtils {
             return qb;
 
         // apply all options from FindOptions
+        if (options.select) {
+            qb.select(options.select.map(selection => qb.alias + "." + selection));
+            qb.partialSelect = true;
```

Comment:
If you check, that's exactly what I'm doing. I'm only selecting the primary columns if they have not already been selected. However it's not enough.

Also, you're coming off as a bit rude you know? I have been courteous and willing to listen to every suggestion.